src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Attributes.cs

```diff
@@ -387,7 +387,24 @@ private static bool ExcludeSpecialAttribute(ICustomAttribute c)
                 case "System.Reflection.AssemblyDelaySignAttribute": return true;
                 case "System.Runtime.CompilerServices.ExtensionAttribute": return true;
                 case "System.Runtime.CompilerServices.DynamicAttribute": return true;
-            }
+                case "System.Runtime.CompilerServices.IsByRefLikeAttribute": return true;
+                case "System.Runtime.CompilerServices.IsReadOnlyAttribute": return true;
+                case "System.ObsoleteAttribute":
+                    {
+                        var arg = c.Arguments.OfType<IMetadataConstant>().FirstOrDefault();
+
+                        if (arg != null && arg.Value is string)
```

Comment:
Nit: Could be shortened to:

```c#
case "System.ObsoleteAttribute":
    return c.Arguments.OfType<IMetadataConstant>().FirstOrDefault()?.Value is string argValue &&
        argValue == "Types with embedded references are not supported in this version of your compiler.";
```

Reply:
I'm not going turn it all the way into a statement like you have but I will use a few of the shortened code fragments.